client/rpcserver/rpcserver_test.go

```diff
@@ -291,6 +291,44 @@ func TestMain(m *testing.M) {
 	os.Exit(doIt())
 }
 
+func TestConnectStart(t *testing.T) {
+	s, _, shutdown := newTServer(t, false, "", "")
+	defer shutdown()
```

Comment:
The race you are seeing makes sense the shutdown func returned by newTServer when start=false does not wait for the server to shutdown.  You'd have to s.Disconnect after s.Connect has succeeded, but not if fails.